src/js/helpers/generalHelpers.ts

```diff
@@ -18,30 +18,35 @@ export const getFieldValueFromKeyString = <T extends Value>(
 };
 
 const setDeepValue = <T extends Value>(
-    value: Value, fieldNames: string[], currentIndex: number, oldValues: Value
+    value: Value, fieldNames: string[], currentIndex: number, oldValues: Value,
 ): T => {
-    if (typeof oldValues !== 'object' && !Array.isArray(oldValues)) {
-        throw new Error('Could not find value for ' + fieldNames.join('.'));
+    let ret;
+    if (typeof oldValues !== 'object') {
+        ret = Number.isNaN(Number(fieldNames[currentIndex])) ? {} : [];
+    } else {
+        ret = Array.isArray(oldValues) ? [...oldValues] : {...oldValues};
     }
 
-    const ret = Array.isArray(oldValues) ? [...oldValues] : {...oldValues};
-
     // any casts because we either index a record with a string or an array with a string that is implicitly cast to an int.
     // this is hard to teach ts
     if (currentIndex === fieldNames.length - 1) {
         (ret as any)[fieldNames[currentIndex]] = value;
-    } else if ((oldValues as any)[fieldNames[currentIndex]] === undefined) {
-        (ret as any)[fieldNames[currentIndex]] = setDeepValue(value, fieldNames, currentIndex + 1, {});
+    } else if (typeof (oldValues as any)[fieldNames[currentIndex]] !== 'object') {
+        if (Number.isNaN(Number(fieldNames[currentIndex + 1]))) {
+            (ret as any)[fieldNames[currentIndex]] = setDeepValue(value, fieldNames, currentIndex + 1, {});
+        } else {
+            (ret as any)[fieldNames[currentIndex]] = setDeepValue(value, fieldNames, currentIndex + 1, []);
+        }
     } else {
         (ret as any)[fieldNames[currentIndex]] = setDeepValue(
-            value, fieldNames, currentIndex + 1, (oldValues as any)[fieldNames[currentIndex]]
+            value, fieldNames, currentIndex + 1, (oldValues as any)[fieldNames[currentIndex]],
         );
     }
     return ret as T;
```

Comment:
I think this T-cast might be obsolete now. But using `let ret;` without an initialization, `ret` is typed `any`, I think. And `any` can be assigned to anything, including `T`. This is probably a nitpick tho.

Reply:
but isn't casting it to T more specific than just returning it as any?

Reply:
the header of the function already defines the return type as T. And if that is given, that wins over the type of the returned variable anyway. So if you leave this cast out and typescript doesn't complain about ret not being assignable to T, the return type will be T.

Reply:
When I remove it, typescript complains:
```
TS2322: Type 'Value[] | { toString(): string; toDateString(): string; toTimeString(): string; toLocaleString(): string; toLocaleString(locales?: string | string[] | undefined, options?: DateTimeFormatOptions | undefined): string; toLocaleString(locales?: LocalesArgument, options?: DateTimeFormatOptions | undefined): string; ... 4...' is not assignable to type 'T'.   'Value[] | { toString(): string; toDateString(): string; toTimeString(): string; toLocaleString(): string; toLocaleString(locales?: string | string[] | undefined, options?: DateTimeFormatOptions | undefined): string; toLocaleString(locales?: LocalesArgument, options?: DateTimeFormatOptions | undefined): string; ... 4...' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'Value'.     Type 'Value[]' is not assignable to type 'T'.       'Value[]' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'Value'.
```

Reply:
Darn... Okay. Then just leave the typecast in. I think this isn't easily solvable without major changes of the types here :/ Gonna have a look at this separately from this PR sometime.